src/osx/cocoa/textctrl.mm

```diff
@@ -1210,6 +1234,26 @@ - (void)controlTextDidEndEditing:(NSNotification *)aNotification
                 break;
         }
     }
+    if( style.HasLeftIndent() )
+    {
+        if( start == end && start == 0 )
```

Comment:
Without speaking about the style (`if (`...), why does this code unconditionally resets `range` after initializing it above and why does it handle `(0, 0)` and `(end, end)` specially?